internal/btf/btf.go

```diff
@@ -176,16 +200,52 @@ func LoadKernelSpec() (*Spec, error) {
 }
 
 func loadKernelSpec() (*Spec, error) {
-	fh, err := os.Open("/sys/kernel/btf/vmlinux")
-	if os.IsNotExist(err) {
-		return nil, fmt.Errorf("can't open kernel BTF at /sys/kernel/btf/vmlinux: %w", ErrNotFound)
-	}
+	release, err := unix.KernelRelease()
 	if err != nil {
-		return nil, fmt.Errorf("can't read kernel BTF: %s", err)
+		return nil, err
+	}
+
+	locations := []struct {
+		pathFmt string
+		raw     bool
+	}{
+		/* try canonical vmlinux BTF through sysfs first */
+		{"/sys/kernel/btf/vmlinux", true /* raw BTF */},
+		/* fall back to trying to find vmlinux ELF on disk otherwise */
+		{"/boot/vmlinux-%s", false},
+		{"/lib/modules/%s/vmlinux-%s", false},
+		{"/lib/modules/%s/build/vmlinux", false},
+		{"/usr/lib/modules/%s/kernel/vmlinux", false},
+		{"/usr/lib/debug/boot/vmlinux-%s", false},
+		{"/usr/lib/debug/boot/vmlinux-%s.debug", false},
+		{"/usr/lib/debug/lib/modules/%s/vmlinux", false},
+	}
+
+	var spec *Spec
+	for _, loc := range locations {
+		path := fmt.Sprintf(loc.pathFmt, release)
+		if err := syscall.Access(path, linux.R_OK); err != nil {
+			continue
+		}
+
+		fh, err := os.Open(path)
+		if err != nil {
+			continue
+		}
+		defer fh.Close()
+		if loc.raw {
+			spec, err = loadNakedSpec(fh, internal.NativeEndian, nil, nil)
+		} else {
+			spec, err = loadNakedSpecFromReader(fh, nil, nil)
+		}
+
+		if err != nil {
+			continue
+		}
+		return spec, nil
 	}
-	defer fh.Close()
 
-	return loadNakedSpec(fh, internal.NativeEndian, nil, nil)
+	return nil, ErrNotFound
```

Comment:
Why ErrNotFound, not `ErrNotSupported` like in the doc of `LoadKernelSpec`? Also, please wrap the error so that callers are forced to use `errors.Is` instead of just comparing.

Reply:
This function was returning `ErrNotFound` previously, so I maintained that. I can change it.